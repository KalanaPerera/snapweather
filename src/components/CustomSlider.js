import React, { Component } from "react";
import Slider from "react-slick";
import { Weather } from "./WeatherCard";

import { connect } from "react-redux";
import { loadWeatherData } from "../actions/";

class CustomSlider extends Component {
  componentDidMount() {
    const { sliderId, list } = this.props;
    this.props.loadWeatherData(list[0], sliderId);
  }


  settings = {
    dots: true,
    lazyLoad: true,
    infinite: true,
    arrows: false,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    initialSlide: 0,
    beforeChange: (current, next) => {
      const { list, sliderId } = this.props;
      this.props.loadWeatherData(list[next], sliderId);
    }
  };

  render() {
    const { sliderId, weather } = this.props;
    const weatherData = weather[sliderId];
    return (
      <div className="col-md-4">
        <Slider {...this.settings}>
          {this.props.list.map(key => (
            <Weather
              theme={this.props.theme}
              weather={weatherData}
              key={key}
              sliderId={sliderId}
              city={key}
            />
          ))} 
        </Slider>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    weather: state.rootReducer
  };
}

export default connect(
  mapStateToProps,
  { loadWeatherData }
)(CustomSlider);
