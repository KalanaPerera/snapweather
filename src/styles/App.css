@import url('https://fonts.googleapis.com/css?family=Muli:400,600,700');

html,
body {
  background-color: #646669;
  font-family: 'Muli', sans-serif;


}

.col-lg-4.component-main-weather {
  width: 33%;
  padding-right: 30px;
}

.component-weather {
  background-color: #fff !important;
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  border-radius: 10px;
}
.component-weather .header {
  padding: 15px;
}

.component-weather .body {
  text-align: center;
  height: 100px;
}

.component-weather .details {
  padding: 10px;
  margin-top: 25px;
  color: #fff;
  text-align: left;
  padding: 15px 0px 15px 0px;
}

.component-weather .details .border-left {
  border-left: 1px solid rgba(121, 137, 142, 0.5);
}

.component-weather .bottom-details {
  background-color: #2a3032;
  color: #d7d7d7;
  height: 110px;
  -webkit-border-bottom-right-radius: 10px;
  -webkit-border-bottom-left-radius: 10px;
  -moz-border-radius-bottomright: 10px;
  -moz-border-radius-bottomleft: 10px;
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;

  text-align: center;
}

.component-weather .bottom-details .set {
  display: table;
  margin: 0 auto;
  padding: 25px 0px 25px 0px;
}

.component-weather .bottom-details .list {
  float: left;
  margin-right: 20px;
  display: inline-flex;
}

.component-weather .bottom-details .list span {
  margin-left: 7px;
  font-size: 14px;
  margin-top: 3px;
}

/* Themes by Weather Category */

.theme-cloudy-sky {
  background: #15c8d1;
}

.theme-heavy-rain {
  background: #ff6c5e;
}

.theme-green-day {
  background: #29df99;
}

/* Helper Styles */

.cursor-pointer{
  cursor: pointer;
}

.bold{
  font-weight: 600
}

.font-12{
  font-size: 12px;
}

.font-20 {
  font-size: 20px;
}

.font-25 {
  font-size: 25px;
}

.font-35 {
  font-size: 35px;
}

.block {
  display: block;
}

/* Application Icons */

.custom-icon-cloud {
  background-image: url("./img/cloud.png");
  background-repeat: no-repeat;
  background-position: center;
  display: inline-block;
  height: 100%;
  width: 100%;
}

.custom-icon {
  background-repeat: no-repeat;
  display: inline-block;
  height: 24px;
  width: 24px;
}

.i-wind {
  background-image: url("./img/wind.png");
}

.i-rain-drop {
  background-image: url("./img/raindrop.png");
}

.i-sun {
  background-image: url("./img/sun.png");
}

.i-moon{
  background-image: url("./img/moon.png");
}

/* Custom Slick Dots */

.slick-dots {
  bottom: 10px;
}

.slick-dots li button:before {
  font-size: 10px;
  color: #fff;
}

.slick-dots li.slick-active button:before {
  opacity: 1;
  color: #fff;
}
