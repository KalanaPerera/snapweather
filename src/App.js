import React, { Component } from "react";
import CustomSlider from "./components/CustomSlider";
import "./styles/App.css";

const sliderMapping = [
  { list: ["Singapore", "Paris", "Dubai"], theme: "theme-cloudy-sky" },
  { list: ["Colombo", "Canada", "Manila"], theme: "theme-heavy-rain" },
  { list: ["India", "Austin", "California"], theme: "theme-green-day" }
];

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="container">
          <div className="row" style={{ marginTop: `100px` }}>
            {sliderMapping.map((slider, idx) => {
              return <CustomSlider {...slider} sliderId={idx} key={idx} />;
            })}
          </div>
        </div>
      </div>
    );
  }
}

export default App;
