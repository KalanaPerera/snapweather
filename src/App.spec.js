import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import CustomSlider from "./components/CustomSlider";

describe("App", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<App />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it("should render the CustomSlider Component", () => {
    expect(wrapper.containsMatchingElement(<CustomSlider />)).toEqual(true);
  });
});
