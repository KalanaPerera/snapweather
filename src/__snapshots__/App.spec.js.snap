// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
          style={
            Object {
              "marginTop": "100px",
            }
          }
        >
          <Connect(CustomSlider)
            list={
              Array [
                "Singapore",
                "Paris",
                "Dubai",
              ]
            }
            sliderId={0}
            theme="theme-cloudy-sky"
          />
          <Connect(CustomSlider)
            list={
              Array [
                "Colombo",
                "Canada",
                "Manila",
              ]
            }
            sliderId={1}
            theme="theme-heavy-rain"
          />
          <Connect(CustomSlider)
            list={
              Array [
                "India",
                "Austin",
                "California",
              ]
            }
            sliderId={2}
            theme="theme-green-day"
          />
        </div>
      </div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
          style={
            Object {
              "marginTop": "100px",
            }
          }
        >
          <Connect(CustomSlider)
            list={
              Array [
                "Singapore",
                "Paris",
                "Dubai",
              ]
            }
            sliderId={0}
            theme="theme-cloudy-sky"
          />
          <Connect(CustomSlider)
            list={
              Array [
                "Colombo",
                "Canada",
                "Manila",
              ]
            }
            sliderId={1}
            theme="theme-heavy-rain"
          />
          <Connect(CustomSlider)
            list={
              Array [
                "India",
                "Austin",
                "California",
              ]
            }
            sliderId={2}
            theme="theme-green-day"
          />
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(CustomSlider)
              list={
                Array [
                  "Singapore",
                  "Paris",
                  "Dubai",
                ]
              }
              sliderId={0}
              theme="theme-cloudy-sky"
            />,
            <Connect(CustomSlider)
              list={
                Array [
                  "Colombo",
                  "Canada",
                  "Manila",
                ]
              }
              sliderId={1}
              theme="theme-heavy-rain"
            />,
            <Connect(CustomSlider)
              list={
                Array [
                  "India",
                  "Austin",
                  "California",
                ]
              }
              sliderId={2}
              theme="theme-green-day"
            />,
          ],
          "className": "row",
          "style": Object {
            "marginTop": "100px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "list": Array [
                "Singapore",
                "Paris",
                "Dubai",
              ],
              "sliderId": 0,
              "theme": "theme-cloudy-sky",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "list": Array [
                "Colombo",
                "Canada",
                "Manila",
              ],
              "sliderId": 1,
              "theme": "theme-heavy-rain",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "list": Array [
                "India",
                "Austin",
                "California",
              ],
              "sliderId": 2,
              "theme": "theme-green-day",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
            style={
              Object {
                "marginTop": "100px",
              }
            }
          >
            <Connect(CustomSlider)
              list={
                Array [
                  "Singapore",
                  "Paris",
                  "Dubai",
                ]
              }
              sliderId={0}
              theme="theme-cloudy-sky"
            />
            <Connect(CustomSlider)
              list={
                Array [
                  "Colombo",
                  "Canada",
                  "Manila",
                ]
              }
              sliderId={1}
              theme="theme-heavy-rain"
            />
            <Connect(CustomSlider)
              list={
                Array [
                  "India",
                  "Austin",
                  "California",
                ]
              }
              sliderId={2}
              theme="theme-green-day"
            />
          </div>
        </div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
            style={
              Object {
                "marginTop": "100px",
              }
            }
          >
            <Connect(CustomSlider)
              list={
                Array [
                  "Singapore",
                  "Paris",
                  "Dubai",
                ]
              }
              sliderId={0}
              theme="theme-cloudy-sky"
            />
            <Connect(CustomSlider)
              list={
                Array [
                  "Colombo",
                  "Canada",
                  "Manila",
                ]
              }
              sliderId={1}
              theme="theme-heavy-rain"
            />
            <Connect(CustomSlider)
              list={
                Array [
                  "India",
                  "Austin",
                  "California",
                ]
              }
              sliderId={2}
              theme="theme-green-day"
            />
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(CustomSlider)
                list={
                  Array [
                    "Singapore",
                    "Paris",
                    "Dubai",
                  ]
                }
                sliderId={0}
                theme="theme-cloudy-sky"
              />,
              <Connect(CustomSlider)
                list={
                  Array [
                    "Colombo",
                    "Canada",
                    "Manila",
                  ]
                }
                sliderId={1}
                theme="theme-heavy-rain"
              />,
              <Connect(CustomSlider)
                list={
                  Array [
                    "India",
                    "Austin",
                    "California",
                  ]
                }
                sliderId={2}
                theme="theme-green-day"
              />,
            ],
            "className": "row",
            "style": Object {
              "marginTop": "100px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "list": Array [
                  "Singapore",
                  "Paris",
                  "Dubai",
                ],
                "sliderId": 0,
                "theme": "theme-cloudy-sky",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "list": Array [
                  "Colombo",
                  "Canada",
                  "Manila",
                ],
                "sliderId": 1,
                "theme": "theme-heavy-rain",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "list": Array [
                  "India",
                  "Austin",
                  "California",
                ],
                "sliderId": 2,
                "theme": "theme-green-day",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
